app/router.js

```diff
@@ -7,16 +7,18 @@ const Router = Ember.Router.extend({
 });
 
 Router.map(function() {
-    this.route('preprints', { path: '/preprints/:file_id' }, function() {
-        this.route('view', { path: '/' });
-        this.route('edit', { path: '/edit' });
-    });
+  this.route('preprints', { path: '/preprints/:file_id' }, function() {
+      this.route('view', { path: '/' });
+      this.route('edit', { path: '/edit' });
+  });
 
-    this.route('add-preprint', { path: '/preprints/add' });
+  this.route('add-preprint', { path: '/preprints/add' });
 
-    this.route('browse');
+  this.route('browse');
 
-    this.route('page-not-found', { path: '/*wildcard' });
+
+  this.route('page-not-found', { path: '/*wildcard' });
+  this.route('discover');
```

Comment:
Conventionally, the wildcard route appears last.

(some routing systems will return the view for the first pattern that matches, in which case it would match the 404 page before discover)